controllers/drplacementcontrol_controller.go

```diff
@@ -1345,6 +1345,20 @@ func (r *DRPlacementControlReconciler) finalizePlacement(
 	placementObj client.Object,
 	log logr.Logger,
 ) error {
+	// TODO: Find a better way to get the home cluster.
+	if drpc.Status.PreferredDecision.ClusterName == "" {
+		return fmt.Errorf("cannot finalize User PlacementRule/Placement: empty preferred decision")
+	}
```

Comment:
Turns out that preferred decision is not updated after failover so we get wrong placemnet:

```
  preferredDecision:
    clusterName: dr1
    clusterNamespace: dr1
```

Need to replace with the way we used to find the home cluster.

Reply:
Current code dos not have this problem.